x11-libs/wxGTK/wxGTK-3.0.5.1.ebuild

```diff
@@ -21,7 +21,9 @@ S="${WORKDIR}/wxWidgets-${PV}"
 LICENSE="wxWinLL-3 GPL-2 doc? ( wxWinFDL-3 )"
 SLOT="${WXRELEASE}"
 KEYWORDS="~alpha amd64 arm arm64 ~hppa ~ia64 ~mips ppc ppc64 ~riscv sparc x86 ~amd64-linux ~x86-linux"
-IUSE="+X doc debug gstreamer libnotify opengl pch sdl tiff webkit"
+IUSE="+X doc debug gstreamer libnotify opengl sdl test tiff webkit"
+REQUIRED_USE="gstreamer? ( X ) libnotify? ( X ) opengl? ( X ) tiff? ( X ) webkit? ( X ) test? ( tiff )"
```

Comment:
Where's this coming from? I'm not claiming it's wrong , but the non-test additions appear new and at least warrant a separate commit with separate explanation? And if so, I can review that properly from a separate commit. Though first thoughts on it are that it's technically correct, but could be bothersome to the user who wants to disable wxGTK and just get wxCore, as needing to disable a bunch of extra flags that could just be no-op then as before. However if we don't add this kind of REQUIRED_USE, we should have all the consumers USE-depping on any of these actually USE dep on both what it already does AND wxGTK[X].

Reply:
After re-researching why I did this originally, it is also to reflect the automagic in the original configure script.  It's a little hard to read due to the indenting where they do things like:

```
if test "$wxUSE_MEDIACTRL" = "yes" -o "$wxUSE_MEDIACTRL" = "auto"; then
    USE_MEDIA=1

                if test "$wxUSE_GTK" = 1; then
        wxUSE_GSTREAMER="no"

                                                                GST_VERSION_MAJOR=1
        GST_VERSION_MINOR=0
        GST_VERSION=$GST_VERSION_MAJOR.$GST_VERSION_MINOR

        if test "$wxUSE_GSTREAMER8" = "no"; then

pkg_failed=no
{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for GST" >&5
$as_echo_n "checking for GST... " >&6; }
```

which basically ignores `--with-gstreamer` if `--without-gtk` is passed.

Reply:
My points about no description in commit log and it really making sense to have as a separate commit remain though. Maybe you can split it into a separate commit and then we can handle them separately and at least cherry-pick the test stuff into main tree until we can figure out which way we want to go with the REQUIRED_USE related thing (having it, or just restructuring the deps to ignore the USE flags when USE=X isn't set, depending on what the situation with consumers is)

Reply:
> My points about no description in commit log and it really making sense to have as a separate commit remain though. Maybe you can split it into a separate commit and then we can handle them separately and at least cherry-pick the test stuff into main tree until we can figure out which way we want to go with the REQUIRED_USE related thing (having it, or just restructuring the deps to ignore the USE flags when USE=X isn't set, depending on what the situation with consumers is)

At least the `tiff? ( X )` part is required in this patch due to `test? ( tiff )` being necessary.